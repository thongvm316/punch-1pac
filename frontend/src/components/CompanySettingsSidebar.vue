<template>
  <ul class="setting-sidebar">
    <li>
      <a
        href="#profile"
        :class="{ active: activeTabName === 'profile' }"
        @click.prevent="setActive('profile')"
      >
        <p-ico-company-profile />
        {{ $t('sidebar.company.profile') }}
      </a>
    </li>
    <li>
      <a
        href="#users"
        :class="{ active: activeTabName === 'users' }"
        @click.prevent="setActive('users')"
      >
        <p-ico-users />
        {{ $t('sidebar.company.users') }}
      </a>
    </li>
    <li>
      <a
        href="#business-days"
        :class="{ active: activeTabName === 'business-days' }"
        @click.prevent="setActive('business-days')"
      >
        <p-ico-business-day />
        {{ $t('sidebar.company.businessDays') }}
      </a>
    </li>
    <li>
      <a
        href="#holidays"
        :class="{ active: activeTabName === 'holidays' }"
        @click.prevent="setActive('holidays')"
      >
        <p-ico-holiday />
        {{ $t('sidebar.company.holidays') }}
      </a>
    </li>
    <li>
      <a
        href="#allowed-ips"
        :class="{ active: activeTabName === 'allowed-ips' }"
        @click.prevent="setActive('allowed-ips')"
      >
        <p-ico-allowed-ip />
        {{ $t('sidebar.company.allowedIPs') }}
      </a>
    </li>
  </ul>
</template>

<script>
import activeTab from '../mixins/active-tab'
import PIcoCompanyProfile from '@/punch-ui/punch-icons/PIcoCompanyProfile'
import PIcoUsers from '@/punch-ui/punch-icons/PIcoUsers'
import PIcoBusinessDay from '@/punch-ui/punch-icons/PIcoBusinessDay'
import PIcoHoliday from '@/punch-ui/punch-icons/PIcoHoliday'
import PIcoAllowedIp from '@/punch-ui/punch-icons/PIcoAllowedIp'

export default {
  name: 'CompanySettingsSidebar',

  components: {
    PIcoCompanyProfile,
    PIcoUsers,
    PIcoBusinessDay,
    PIcoHoliday,
    PIcoAllowedIp
  },

  mixins: [activeTab]
}
</script>
