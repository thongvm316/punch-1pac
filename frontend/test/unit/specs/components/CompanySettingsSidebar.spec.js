import { shallowMount } from '@vue/test-utils'

import wrapperOps from '../../supports/wrapper'

import CompanySettingsSidebar from '@/components/CompanySettingsSidebar'

const wrapper = shallowMount(CompanySettingsSidebar, wrapperOps)

describe('CompanySettingsSidebar.vue', () => {
  it('should render correctly', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
})
