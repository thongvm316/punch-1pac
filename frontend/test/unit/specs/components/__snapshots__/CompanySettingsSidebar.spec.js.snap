// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanySettingsSidebar.vue when mounted should render correctly 1`] = `
<ul class="setting-sidebar">
  <li class="active">
    <picocompanyprofile-stub></picocompanyprofile-stub>
    Profile
  </li>
  <li class="">
    <picousers-stub></picousers-stub>
    Users
  </li>
  <li class="">
    <picobusinessday-stub></picobusinessday-stub>
    Business Days
  </li>
  <li class="">
    <picoholiday-stub></picoholiday-stub>
    Holidays
  </li>
  <li class="">
    <picoallowedip-stub></picoallowedip-stub>
    Allowed IPs
  </li>
</ul>
`;

exports[`CompanySettingsSidebar.vue when setActive should active Holidays tab 1`] = `
<ul class="setting-sidebar">
  <li class="">
    <picocompanyprofile-stub></picocompanyprofile-stub>
    Profile
  </li>
  <li class="">
    <picousers-stub></picousers-stub>
    Users
  </li>
  <li class="">
    <picobusinessday-stub></picobusinessday-stub>
    Business Days
  </li>
  <li class="active">
    <picoholiday-stub></picoholiday-stub>
    Holidays
  </li>
  <li class="">
    <picoallowedip-stub></picoallowedip-stub>
    Allowed IPs
  </li>
</ul>
`;
